tools/wpt/expectation.json

```diff
@@ -6908,6 +6908,9 @@
         }
       },
       "embedded-content": {
+        "the-canvas-element": {
+          "imagedata.html": true
```

Comment:
This test case does not run with Deno's current wpt set up. However, by hacking the `wpt.ts` script I got the test running and it appears to pass.

https://github.com/web-platform-tests/wpt/blob/360d99ddc756540791ed1e1e97df51fa950aa38d/html/semantics/embedded-content/the-canvas-element/imagedata.html:

```
/html/semantics/embedded-content/the-canvas-element/imagedata.html

test ImageData(w, h), width cannot be 0 ... ok
test ImageData(w, h), height cannot be 0 ... ok
test ImageData(w, h), exposed attributes check ... ok
test ImageData(buffer, w), the buffer size must be a multiple of 4 ... ok
test ImageData(buffer, w), buffer size must be a multiple of the image width ... ok
test ImageData(buffer, w, h), buffer.length == 4 * w * h must be true ... ok
test ImageData(buffer, w, opt h), Uint8ClampedArray argument type check ... ok
test ImageData(buffer, w, opt h), exposed attributes check ... ok

file result: ok. 8 passed; 0 failed; 0 expected failure; total 8 (2681ms)
```

Reply:
I believe we have a few hacks in place for WPT. What is specifically failing?

Reply:
the file name is the issue @mmastrac.

To fix this, an upstream PR to WPT should be made to update the naming convention for this specific testing module.

Reply:
> I believe we already have a few hacks in place for WPT. What is specifically failing?

When running `./tools/wpt.ts run` the test case is not discovered and not run by the script.

> To fix this, an upstream PR to WPT should be made to update the naming convention for this specific testing module

Is this really the case? I might be misunderstanding the situation, but test and name seems valid to me. The test is the "With HTML boilerplate" type according to [the documentation](https://web-platform-tests.org/writing-tests/testharness.html#with-html-boilerplate). It appears that these types of tests have not been used yet by Deno.

Do we need to refactor the [discoverTestsToRun](https://github.com/jamsinclair/deno/blob/2665ca103e64ae07d7c29d3400d0c37ec691ff50/tools/wpt.ts#L694) to support them?

Reply:
I've added the commit [28347cf](https://github.com/denoland/deno/pull/21183/commits/28347cf61fd62560093f22762498c1d0e2a83c0d) which I believe resolves the issue.

Reply:
I see, you are correct. in previous cases we have gone upstream to update the HTML tests to js tests where applicable

Reply:
Ah I see! Well I have no strong feelings, I'll follow whatever your team prefers 🙇